src/DistributedMetadata/PlacementStrategy.cpp

```diff
@@ -16,6 +18,18 @@ std::vector<NodeMetricsPtr> DiskStrategy::qualifiedNodes(const NodeMetricsContai
 
     for (const auto & [node, metrics] : nodes_metrics)
     {
+        /// If the role of the node is "ingest" or "search", skip it.
```

Comment:
I think `node.roles.find("search")` `node.roles.find("ingest")` is good to go. We don't need this complexity